---
title: "GCGA vs PBS"
output: html_document
---

```{r setup, include=TRUE, message=FALSE, warning=FALSE}
knitr::opts_chunk$set(echo = TRUE, message = FALSE)
library(tximeta)
library(tidyverse)
library(pheatmap)
library(RColorBrewer)
library(ggrepel)
library(ggplot2)
library(hexbin)
library(SummarizedExperiment)
library(DESeq2)
library(AnnotationDbi)
library(org.Mm.eg.db)
library(clusterProfiler)
library(repr)
library(writexl)
```

### Index

An index is generated with files from: [Index files](https://www.gencodegenes.org/mouse/). The following code were used to download the most recent files: 
```{r eval=FALSE, echo=TRUE}
wget ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_mouse/release_M28/gencode.vM28.transcripts.fa.gz
wget ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_mouse/release_M28/GRCm39.primary_assembly.genome.fa.gz
```

Metadata is prepared:
```{r eval=FALSE, echo=TRUE}
grep "^>" <(gunzip -c GRCm39.primary_assembly.genome.fa.gz) | cut -d " " -f 1 > decoys.txt
sed -i.bak -e 's/>//g' decoys.txt
```

The transcriptome and genome are concatenated:
```{r eval=FALSE, echo=TRUE}
cat gencode.vM28.transcripts.fa.gz GRCm39.primary_assembly.genome.fa.gz > gentrome.fa.gz
```

##### Index command:[^1]

The decoy-aware index is created using salmon:
```{r eval=FALSE, echo=TRUE}
salmon index -t gentrome.fa.gz -d decoys.txt -p 20 -i salmon_index --gencode
```

[^1]: [Site for index knowlegde](https://combine-lab.github.io/alevin-tutorial/2019/selective-alignment/)

### FastQC

The following command was used to generated FastQC files for each fastq file:
```{r eval=FALSE, echo=TRUE}
fastqc /*.fastq.gz -o /FastQC_Reports
```

### Mapping

After FASTQC the following command was used for selective alignment with Salmon:
```{r eval=FALSE, echo=TRUE}
for i in $(ls ~/ | sed s/_R1.fastq.gz// |sort -u | grep -v '_R2');do echo ${i}; salmon quant -i ~/salmon_index/  --libType A --seqBias -- gcBias  -1 ~/${i}_R1.fastq.gz -2 ~/${i}_R2.fastq.gz --validateMappings -p 20 -o ~/Salmon_output/${i}; done 
```

### Data and study design import

From Salmon transcripts/isoforms can be extracted with *tximeta* and combined with the study design data.
```{r}
csvfile <-
  file.path(
    "~/",
    "study_design_GCGA.csv"
  )
coldata <- read.csv(csvfile, stringsAsFactors = FALSE)
coldata$names <- coldata$Sample
coldata$files <-
  file.path(
    "~/",
    coldata$names,
    "quant.sf"
  )

head(coldata, 10)
```

The Summarized Experiment object is created:
```{r warning=FALSE}
se <- tximeta(coldata, type = "salmon")
dim(se)
```

140790 transcripts across the 15 samples are summarized to gene level with *tximeta*: 
```{r message=FALSE, warning=FALSE}
gse <- summarizeToGene(se)
dim(gse)
```
There is 54307 genes across the 15 samples.

The library sizes are accessed:
```{r}
round(colSums(assay(gse)) / 1e6)
```

A DESeq2 object with comparisons made across the *Treatment* is created. 
```{r warning=FALSE}
dds <- DESeqDataSet(gse, design = ~ Treatment)
dds
```

In order to reduce the noise in the data, we removed the rows that have no or nearly no information about the amount of gene expression. Here, we remove rows of the DESeqDataSet that have no counts, or only a single count across all samples.
```{r}
nrow(dds)
dds <- dds[rowSums(counts(dds)) > 1,]
nrow(dds)
```
24647 genes are left across the 15 samples.

Since we are not interested in pseudogenes, these are likewise removed from further analysis:
```{r}
ens.str <- substr(rownames(dds), 1, 18)
ens.str_full <- substr(rownames(dds), 1, 21)
df_pseudo <-
  as.data.frame(ens.str) %>% add_column(as.data.frame(ens.str_full))
df_pseudo$symbol <- mapIds(
  org.Mm.eg.db,
  keys = ens.str,
  column = "SYMBOL",
  keytype = "ENSEMBL",
  multiVals = "first"
)
df_pseudo <-
  as_tibble(filter(df_pseudo,!is.na(symbol)))  %>% column_to_rownames("ens.str_full")
pseudo_remove <- substr(rownames(df_pseudo), 1, 21)

dds <- dds[pseudo_remove, ]
nrow(dds)
```
18771 genes across 15 samples are left in the dataset.

### Exploratory Data Analysis

##### Normalization

We use the algorithm *variance stabilizing transformation* (VST) for negative binomial data with a dispersion-mean trend (Anders and Huber 2010) offered by the DESeq2 package. 
```{r fig.align='center', message=FALSE, warning=FALSE}
vst <- vst(dds, blind = FALSE)
```

##### Sample similarity

```{r}
sampleDists <- dist(t(assay(vst)))
sampleDists
```

```{r, fig.align='center'}
sampleDistMatrix <- as.matrix(sampleDists)
rows <- rownames(sampleDistMatrix)
rownames(sampleDistMatrix) <- NULL
colors <- colorRampPalette(rev(brewer.pal(9, "Blues")))(255)
Treatment <- dds@colData@listData[["Treatment"]]
anno <- data.frame(Treatment, row.names = dds@colData@rownames)
annotation_colors <- list(Treatment = c("GCGA" = "#FF6600", "PBS" = "black"))
pheatmap(
  sampleDistMatrix,
  annotation_col = anno,
  annotation_colors = annotation_colors,
  clustering_distance_rows = sampleDists,
  clustering_distance_cols = sampleDists,
  clustering_method = "complete",
  col = colors,
  main = "Heatmap with hierarchical clustering",
  height = 10,
  width = 5
)
```

##### Principal component analysis

```{r, fig.align='center'}
pcaData <-
  plotPCA(vst,
          intgroup = c("Treatment", "Sample"),
          returnData = TRUE)
percentVar <- round(100 * attr(pcaData, "percentVar"))

ggplot(pcaData, aes(
  x = PC1,
  y = PC2,
  color = Treatment,
  label = Sample
)) +
  geom_point(size = 3) +
  ggrepel::geom_text_repel(max.overlaps = 20) +
  scale_color_manual(values = c("#FF6600", "black")) +
  xlab(paste0("PC1: ", percentVar[1], "% variance")) +
  ylab(paste0("PC2: ", percentVar[2], "% variance")) +
  coord_fixed() +
  ggtitle("PCA with vst normalized data") +
  theme_bw()
```

Sample 8 seems like an outlier and is removed from the dataset:
```{r}
vst_idx <- grep("8", colnames(vst))
rvst <- vst[, -vst_idx]
```

The PCA plot is recreated w/o sample 8:
```{r fig.align='center', message=FALSE}
pcaData <-
  plotPCA(rvst,
          intgroup = c("Treatment", "Sample"),
          returnData = TRUE)
percentVar <- round(100 * attr(pcaData, "percentVar"))

ggplot(pcaData, aes(
  x = PC1,
  y = PC2,
  color = Treatment,
  label = Sample
)) +
  geom_point(size = 3) +
  ggrepel::geom_text_repel(max.overlaps = 20) +
  scale_color_manual(values = c("#FF6600", "black")) +
  xlab(paste0("PC1: ", percentVar[1], "% variance")) +
  ylab(paste0("PC2: ", percentVar[2], "% variance")) +
  coord_fixed() +
  theme_bw() +
  theme(legend.position = "top") +
  theme(legend.title = element_blank())
```

###### Removing sample 8

As sample 8 is a clear outlier, this sample is removed from the downstream analysis.
```{r}
idx <- grep("8", colnames(dds))
dds <- dds[, -idx]
dds
```

### DE analysis

Analysis between the two groups are made:
```{r}
dds <- DESeq(dds)

GCGA_PBS <- results(
  dds,
  contrast = c("Treatment", "GCGA", "PBS"),
  lfcThreshold = 0,
  alpha = 0.05
)
```

A summary of the analysis is printed:
```{r}
summary(GCGA_PBS)
```

#### DE diagnostic plots

In this section, diagnostic plots are created to ensure that the differential expression analysis is appropriate for the data.

##### Dispersion plot

```{r, fig.align='center'}
plotDispEsts(dds)
```

##### MA plot

An MA plot is created to observe if the normalization was done appropriately. 
```{r fig.align='center', message=FALSE, warning=FALSE}
GCGA_PBS_tibble <- GCGA_PBS %>%
  as.data.frame(.) %>%
  rownames_to_column("ENSEMBL") %>%
  as_tibble(.)

ggplot(GCGA_PBS_tibble) +
  geom_point(
    aes(x = baseMean, y = log2FoldChange, color = padj < 0.05),
    alpha = 0.5,
    size = 1
  ) +
  geom_smooth(aes(x = baseMean, y = log2FoldChange)) +
  scale_x_log10() +
  ylim(-10, 10) +
  geom_hline(yintercept = 0,
             alpha = 0.5,
             color = "red") +
  labs(title = "MA-plot (GCGA vs PBS)",
       y = "Log2 fold change",
       x = "Mean of normalized counts") +
  scale_color_discrete(name = "Adjusted p-value",
                       labels = c("> 0.05", "< 0.05", "Not tested")) +
  theme_bw() +
  theme(
    legend.position = c(0.95, 0.95),
    legend.justification = c("right", "top"),
    legend.background = element_rect(linetype = "solid", colour = "black")
  )
```

##### P-value plot

The distribution of p-values is assessed were genes with very small counts are excluded:
```{r, fig.align='center'}
hist(
  GCGA_PBS$pvalue[GCGA_PBS$baseMean > 1],
  breaks = 0:20 / 20,
  col = "grey50",
  border = "white",
  main = "P-value distribution for GCGA vs PBS",
  xlab = "P-value"
)
```

##### Independent filtering plot

```{r fig.align='center', message=FALSE, warning=FALSE}
ggplot(GCGA_PBS_tibble,
       aes(
         x = dense_rank(baseMean),
         y = -log10(pvalue),
         color = padj < 0.05
       )) +
  geom_point(alpha = 0.5) +
  labs(title = "Independent filtering (GCGA vs PBS)",
       y = "- Log10 p-value",
       x = "Base mean expression - Ranked") +
  scale_color_discrete(name = "Adjusted p-value",
                       labels = c("> 0.05", "< 0.05", "Not tested")) +
  theme_bw() +
  theme(
    legend.position = c(0.05, 0.95),
    legend.justification = c("left", "top"),
    legend.background = element_rect(linetype = "solid",
                                     colour = "black")
  )
```

#### DE results

Adding gene symbols and gene names:
```{r message=FALSE, warning=FALSE}
ens.str <- substr(rownames(GCGA_PBS), 1, 18)
GCGA_PBS$symbol <- mapIds(
  org.Mm.eg.db,
  keys = ens.str,
  column = "SYMBOL",
  keytype = "ENSEMBL",
  multiVals = "first"
)
GCGA_PBS$gene_id <- mapIds(
  org.Mm.eg.db,
  keys = ens.str,
  column = "ENTREZID",
  keytype = "ENSEMBL",
  multiVals = "first"
)
GCGA_PBS$gene_name <- mapIds(
  org.Mm.eg.db,
  keys = ens.str,
  column = "GENENAME",
  keytype = "ENSEMBL",
  multiVals = "first"
)

GCGA_PBS_tibble <-
  add_column(GCGA_PBS_tibble,
             gene_symbol = GCGA_PBS$symbol,
             .after = "ENSEMBL")
GCGA_PBS_tibble <-
  add_column(GCGA_PBS_tibble,
             gene_name = GCGA_PBS$gene_name,
             .after = "gene_symbol")
```

All differentially expressed genes (FDR < 0.05) are saved as *All_Sig_Genes_GCGA.xlsx*
```{r}
All_Sig_Genes_GCGA <-
  arrange(GCGA_PBS_tibble, desc(log2FoldChange)) %>%
  filter(., padj < 0.05) %>%
  mutate(., foldChange = 2 ^ log2FoldChange)

write_xlsx(All_Sig_Genes_GCGA, "All_Sig_Genes_GCGA.xlsx")
write_csv(All_Sig_Genes_GCGA, "All_Sig_Genes_GCGA.csv")
```

### Gene Ontology 

Lists for gene ontology analysis are prepared:
```{r}
GO_GCGA_PBS <- GCGA_PBS
genes <-
  lapply(rownames(GO_GCGA_PBS), function(x)
    unlist(strsplit(x, ".", fixed = TRUE))[1])
rownames(GO_GCGA_PBS) <- genes
resSig <- subset(GO_GCGA_PBS, padj < 0.05)
resFCordered <-
  unique(GO_GCGA_PBS[order(GO_GCGA_PBS$log2FoldChange, decreasing = TRUE), ])
sigFCordered <-
  unique(resSig[order(resSig$log2FoldChange, decreasing = TRUE), ])

universe <- resFCordered[, c("log2FoldChange")]

siggenes <-
  sigFCordered[abs(sigFCordered$log2FoldChange) > 0, c("log2FoldChange")]
names(universe) <- rownames(resFCordered)
names(siggenes) <-
  rownames(sigFCordered[abs(sigFCordered$log2FoldChange) > 0, ])

length(universe)
length(siggenes)
```

Significant genes are grouped in up- and down-regulated genes and *compareCluster* is used to analyze enriched pathways: 
```{r}
mydf_GCGA <- data.frame(Ensembl = names(siggenes), FC = siggenes)
mydf_GCGA$Genes <- "Up-regulated"
mydf_GCGA$Genes[mydf_GCGA$FC < 0] <- "Down-regulated"

formula_res <- compareCluster(
  Ensembl ~ Genes,
  data = mydf_GCGA,
  universe = names(universe),
  fun = "enrichGO",
  OrgDb = 'org.Mm.eg.db',
  ont = 'BP',
  readable = TRUE,
  pAdjustMethod = "BH",
  pvalueCutoff  = 0.05,
  keyType = 'ENSEMBL'
)
```

All GOBPs are saved in an excel file:
```{r}
export_formula_res <-
  as_tibble(formula_res) %>% dplyr::select(c(-"Cluster", -"qvalue", -"Count"))

write_xlsx(export_formula_res, "GOBP_Genes_GCGA_PBS.xlsx")
write_csv(export_formula_res, "GOBP_Genes_GCGA_PBS.csv")
```

The lists used for making the GOBP analysis is saved:
```{r}
write.csv(mydf_GCGA, "mydf_GCGA.csv")
write.csv(universe, "universe_GCGA.csv")
```

### Shiny app

A *for* loop creating normalized counts (DESeq2) for all genes and subsequently storing them in a tibble is made:
```{r eval=FALSE, include=TRUE}
t_box_shiny <- tibble(count = 1,
                      Treatment = NA,
                      ENSEMBL = NA)

for (i in rownames(dds)) {
  pCdata_box <-
    plotCounts(dds,
               gene = i,
               intgroup = "Treatment",
               returnData =  TRUE)  %>% as_tibble() %>% add_column(ENSEMBL = i)
  t_box_shiny <- bind_rows(x = t_box_shiny, y = pCdata_box)
}

t_box_shiny <- t_box_shiny[-1, ]
```

Relevant data is saved for a shiny app:
```{r eval=FALSE, include=TRUE}
Export_GCGA_PBS_tibble <- GCGA_PBS_tibble %>%
  mutate(., foldChange = 2 ^ log2FoldChange) %>%
  relocate(., foldChange, .after = log2FoldChange)

write_csv(Export_GCGA_PBS_tibble, file = "~/GCGA_PBS_shiny.csv")
write_csv(t_box_shiny,"~/t_box_shiny.csv")
write_csv(export_formula_res, "~/GOBP_Genes_GCGA_PBS.csv")
```

### Session Info

```{r}
sessionInfo()
```